ui/src/Recipes.tsx

```diff
@@ -1,40 +1,50 @@
 import { useLocation } from "react-router-dom";
 import { useEffect, useState } from "react";
 import { Divider } from "antd";
-import { RecipesModel } from "./models";
 import { Service } from "./service";
 import { useTranslation } from "react-i18next";
+import { Api, Recipe, RecipesListParams } from "./api/nowaste";
 
 // A custom hook that builds on useLocation to parse
 // the query string for you.
 function useQuery() {
   return new URLSearchParams(useLocation().search);
 }
 
+function RecipeTile(props: { recipe: Recipe }) {
+  return (
+    <>
+      <h4>{props.recipe.title}</h4>
+      {props.recipe.imgUrl && (
+        <img
+          src={props.recipe.imgUrl}
+          style={{ width: "100px" }}
+          alt={"recipe photo"}
+        />
+      )}
+    </>
+  );
+}
+
 export function Recipes(): JSX.Element {
+  const api = new Api();
   const { t } = useTranslation();
-  const query = useQuery();
-  const [recipes, setRecipes] = useState<Service<RecipesModel[]>>({
+  const urlQuery = useQuery();
+  const [recipes, setRecipes] = useState<Service<Recipe[]>>({
     status: "init",
   });
-  let queryStr = "";
 
-  if (query.getAll("q")) {
-    queryStr =
-      "?" +
-      query
-        .getAll("q")
-        .map((p) => `q=${p}`)
-        .join("&");
+  const recipesQuery = {} as RecipesListParams;
+
+  if (urlQuery.has("q")) {
+    recipesQuery.q = urlQuery.getAll("q");
   }
 
   useEffect(() => {
     async function fetchRecipes() {
-      const response = await fetch(`/api/recipes${queryStr}`);
+      const response = await api.api.recipesList(recipesQuery);
```

Comment:
The second `api` is auto-generated. I can wrap it in a function to make it more readable.
I hoped this will somehow go through the review. I was more scared about using the root account on the database :laughing: